skimage/registration/tests/test_lddmm.py

```diff
@@ -0,0 +1,377 @@
+from itertools import product
+import numpy as np
+
+from scipy.ndimage import rotate
+from scipy.ndimage import map_coordinates
+
+from skimage.registration._lddmm import diffeomorphic_metric_mapping
+
+"""
+Test diffeomorphic_metric_mapping.
+"""
+
+
+class Test_diffeomorphic_metric_mapping:
+    def _test_diffeomorphic_metric_mapping(
+        self, rtol=0, atol=1 - 1e-9, **diffeomorphic_metric_mapping_kwargs
+    ):
+        """
+        A helper method for this class to verify registrations once they are
+        computed.
+        """
+
+        lddmm_output = diffeomorphic_metric_mapping(
+            **diffeomorphic_metric_mapping_kwargs
+        )
+
+        reference_image = diffeomorphic_metric_mapping_kwargs[
+            "reference_image"
+        ].astype(float)
+        moving_image = diffeomorphic_metric_mapping_kwargs[
+            "moving_image"
+        ].astype(float)
+        reference_image_spacing = (
+            diffeomorphic_metric_mapping_kwargs["reference_image_spacing"]
+            if "reference_image_spacing"
+            in diffeomorphic_metric_mapping_kwargs.keys()
+            else 1
+        )
+        moving_image_spacing = (
+            diffeomorphic_metric_mapping_kwargs["moving_image_spacing"]
+            if "moving_image_spacing"
+            in diffeomorphic_metric_mapping_kwargs.keys()
+            else 1
+        )
+
+        # Applying the transforms using map_coordinates assumes
+        # map_coordinates_ify was left as True.
+
+        deformed_moving_image = map_coordinates(
+            input=moving_image,
+            coordinates=lddmm_output.moving_image_to_reference_image_transform,
+        )
+
+        deformed_reference_image = map_coordinates(
+            input=reference_image,
+            coordinates=lddmm_output.reference_image_to_moving_image_transform,
+        )
+
+        assert np.allclose(
+            deformed_reference_image, moving_image, rtol=rtol, atol=atol
+        )
+        assert np.allclose(
+            deformed_moving_image, reference_image, rtol=rtol, atol=atol
+        )
```

Comment:
This `np.allclose` method with `atol=1 - 1e-9` seems strange to me. The input images are in the range [0, 1] so the tolerance is nearly the full intensity range?

Two other possibilities for comparison are `skimage.metrics.normalized_mutual_information` or `skimage.metrics.normalized_root_mse`. `normalized_root_mse` seems okay as a comparison for these datasets where the intensities of the moving and reference images match.